### ISSUE TITLE
Incorrect UV map used when using Lightmap component

### ISSUE THREAD

**Issue by GitGhillie** [bevyengine/bevy#13122]
## Bevy version

0.13, main

## Relevant system information

```ignore
`AdapterInfo { name: "AMD Radeon RX 7900 XT", vendor: 4098, device: 29772, device_type: DiscreteGpu, driver: "AMD proprietary driver", driver_info: "23.12.1 (LLPC)", backend: Vulkan }`
```

## What you did

See https://github.com/GitGhillie/bake_repro for code and blend files. Basically:
- Create plane and cube in Blender
- Add a simple texture to both (issue also occurs without textures)
- Bake lighting
- Export gltf

## What went wrong

The Bevy render looks awful, probably due to using the wrong UVs:
![image](https://github.com/bevyengine/bevy/assets/23659539/68d3da4b-8770-4b29-9bf9-f81f767fcd74)

Using an online gltf viewer it looks correct, so I'm pretty sure it's something on the Bevy side:
![image](https://github.com/bevyengine/bevy/assets/23659539/271bfa42-180d-478b-a2b3-5300c74cab2b)

## Additional information

Related Discord help thread: https://discord.com/channels/691052431525675048/1224462995875168426/1224462995875168426

**Comment by GitGhillie** [comment]
Looks like this might be a duplicate of https://github.com/bevyengine/bevy/issues/13086, maybe someone with more knowledge on the topic can confirm

**Comment by bugsweeper** [comment]
Got such view with #13153 change
![bake_repro_screenshot](https://github.com/bevyengine/bevy/assets/17225606/17745d4a-11ea-449a-ad1b-2eb854cdd176)

**Comment by GitGhillie** [comment]
I took a closer look using your branch and it works! Thank you for chasing this down

**Comment by bugsweeper** [comment]
@alice-i-cecile this issue should be reopened, after #13402 changes
![image](https://github.com/bevyengine/bevy/assets/17225606/246f40d5-e3d7-4957-b12a-619c227e1afc)

**Comment by geckoxx** [comment]
There is probably something wrong with your glb file. With the gltf file from your repo it is correct. If I generate a new glb from it it is also right.
